### PR TITLE
Regional disk resource policies attachment resource.

### PR DESCRIPTION
**Pull request by ndmckinley**
Fixes https://github.com/terraform-providers/terraform-provider-google/issues/5357.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_compute_region_disk_resource_policy_attachment`
```

### CODE REVIEW

File: templates/terraform/decoders/compute_region_disk_resource_policies_attachment.go.erb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+res["name"] = GetResourceNameFromSelfLink(res["name"].(string))
```
[the remark this answers is not in the document]
Comment (ndmckinley) [reply templates/terraform/decoders/compute_region_disk_resource_policies_attachment.go.erb]: Good call, sure.

File: templates/terraform/encoders/compute_region_disk_resource_policies_attachment.go.erb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+if err != nil {
+  return nil, err
+}
+
+region, err := getRegion(d, config)
+if err != nil {
+	return nil, err
+}
+if region == "" {
+	return nil, fmt.Errorf("zone must be non-empty - set in resource or at provider-level")
```
Comment (danawillow) [inline templates/terraform/encoders/compute_region_disk_resource_policies_attachment.go.erb]: drive-by: s/zone/region?

Comment (ndmckinley) [reply templates/terraform/encoders/compute_region_disk_resource_policies_attachment.go.erb]: Whoops, missed that one.

Comment (ndmckinley) [reply templates/terraform/encoders/compute_region_disk_resource_policies_attachment.go.erb]: https://github.com/GoogleCloudPlatform/magic-modules/pull/3229
